src/main/java/gregtech/api/metatileentity/MetaTileEntity.java

```diff
@@ -811,19 +812,19 @@ private void updateLightValue() {
     }
 
     public void update() {
-        if (!allowTickAcceleration()) {
-            int currentTick = FMLCommonHandler.instance().getMinecraftServerInstance().getTickCounter();
-            if (currentTick == lastTick) {
-                return;
-            }
-            lastTick = currentTick;
-        }
         for (MTETrait mteTrait : this.mteTraits.values()) {
             if (shouldUpdate(mteTrait)) {
                 mteTrait.update();
             }
         }
         if (!getWorld().isRemote) {
+            if (!allowTickAcceleration()) {
```

Comment:
It's important this code happens before updating `MTETrait`s, otherwise things like recipe logic will still get accelerated. I think it should just have its own `!getWorld().isRemote` check and stay at the top.